exercises/dolica_akelloegwel/ex02_oo_basics/src/main.cpp

```diff
@@ -0,0 +1,44 @@
+#include "Square.h"
+#include "Rectangle.h"
+#include "Circle.h"
+#include "Triangle.h"
+#include "ShapeSorter.h"
+
+#ifndef _Shape_h
+#define _Shape_h
+
+int main(int, char **)
+{
+	// Create various shapes
+	double squareSide = 2.5;
+	Square mySquare = Square(squareSide);
+
+	double recWidth = 2;
+	double recHeight = 5;
+	Rectangle myRectangle = Rectangle(recWidth, recHeight);
+
+	double circleRadius = 5;
+	Circle myCircle = Circle(circleRadius);
+
+	double triangleBase = 3;
+	double triangleHeight = 4;
+	Triangle myTriangle = Triangle(triangleBase, triangleHeight);
+
+	// Create a vector and place the shapes in it
+	std::vector<Shape> myShapes;

```

Comment:
For a large vector this could be problematic. The vector will be 24 bytes + n*size_of(Shape). Also copies of vectors will force deep copies of your shape.